src/opm/input/eclipse/Schedule/ArrayDimChecker.cpp

```diff
@@ -32,37 +33,116 @@
+                     const std::size_t           maxVal,
+                     const std::size_t           schedVal,
+                     const std::size_t           item,
+                     std::string_view            hostEntity,
+                     std::string_view            entity,
+                     const std::string&          ctxtKey,
+                     const Opm::ParseContext&    ctxt,
+                     Opm::ErrorGuard&            guard)
+    {
+        using namespace fmt::literals;
+
+        const auto* pl = (maxVal != 1) ? "are" : "is";
+
+        // Note: Number of leading blanks on lines 2-4 affects the formatted output.
+        const auto message_fmt = fmt::format(R"(Problem with keyword {{keyword}}
+  In {{file}} line {{line}}
+  The case has a {hostEntity} with {schedVal} {entity}, but at most {maxVal} {pl} allowed in {{keyword}}.
+  Please increase item {item} of {{keyword}} to at least {schedVal})",
+                                             "schedVal"_a = schedVal,
+                                             "maxVal"_a = maxVal,
+                                             "hostEntity"_a = hostEntity,
+                                             "entity"_a = entity,
+                                             "item"_a = item,
+                                             "pl"_a = pl);
+
+        ctxt.handleError(ctxtKey, message_fmt, location, guard);
+    }
+
     namespace WellDims {
+
         void checkNumWells(const Opm::Welldims&     wdims,
                            const Opm::Schedule&     sched,
                            const Opm::ParseContext& ctxt,
                            Opm::ErrorGuard&         guard)
         {
             const auto nWells = sched.numWells();
 
-            if (nWells > std::size_t(wdims.maxWellsInField()))
-            {
-                const auto& location = wdims.location();
-                if (location) {
-                    std::string fmt_message = fmt::format("Problem with keyword {{keyword}}\n"
-                                                          "In {{file}} line {{line}}\n"
-                                                          "The case has {0} wells - but only {1} are specified in WELLDIMS.\n"
-                                                          "Please increase item 1 in WELLDIMS to at least {1}", nWells, wdims.maxWellsInField());
-
-                    ctxt.handleError(Opm::ParseContext::RUNSPEC_NUMWELLS_TOO_LARGE,
-                                     fmt_message, *location, guard);
-                } else {
-                    std::string msg = fmt::format("The case does not have a WELLDIMS keyword.\n"
-                                                  "Please add a WELLDIMS keyword in the RUNSPEC section specifying at least {} wells in item 1.", nWells);
-                    ctxt.handleError(Opm::ParseContext::RUNSPEC_NUMWELLS_TOO_LARGE, msg, {}, guard);
-                }
```

Comment:
It seems like this error handling now will never take place because of of a plain return (same below. Is that intentional?

Reply:
> It seems like this error handling now will never take place because of a plain return

I'm reversing the logic to reduce the level of nesting.  The original code was
```C++
if (dynamic_size > limit) {
    reportError();
}
```
This logic is
```C++
if (dynamic_size <= limit) {
    // all good
    return;
}
reportError();
```

Reply:
I was actually wondering where the else branch inside of the outer if went. But I now realize that it is there.